### PR TITLE
requestIdleCallback

### PR DESCRIPTION
**Pull request by ronag**
This makes deepstream heavy web applications much more responsive.

### CODE REVIEW

**Comment by ronag** [comment]
Not sure why travis failed. Any ideas?

**Comment by yasserf** [comment]
@ronag Not sure why it is failing.  We spoke about this during our last planning meeting but I want to do a bit more investigation before merging it in. Will keep this updated as I go along.

File: src/message/connection.js [hunk 1]
```diff
@@ -307,23 +321,45 @@ Connection.prototype._onClose = function() {
...
+				this._handleAuthResponse( parsedMessage );
+			}
+			else {
+				this._client._$onMessage( parsedMessage );
+			}
 		}
+	} while( deadline.timeRemaining() > 4 )
+
+	if ((this._parsedMessages.length > 0 || this._rawMessages.length > 0) && !this._deliberateClose) {
+		this._messageHandler = utils.requestIdleCallback( this._handleMessages.bind( this ) );
```
Comment (ronag) [inline src/message/connection.js]: Mission idletimeout option

**Comment by ronag** [comment]
@yasserf: The reason this fails is probably due to #204 and would be fixed by #209.

**Comment by yasserf** [comment]
Hey, can you confirm if you have this in your fork and been using it? Generally makes alot of sense

**Comment by ronag** [comment]
We've been using it for the last 6 months with out issue.

**Comment by ronag** [comment]
This PR is a bit outdated though (https://github.com/nxtedition/deepstream.io-client-js/blob/master/src/message/connection.js). I'll see if I can fix that later today.

**Comment by yasserf** [comment]
That would be great. Sorry for the massive detail in merging this in, I'll do a release today and merge it straight into master for us to play with and do a follow up release.

**Comment by ronag** [comment]
Closed in favor of #340

**Pull request closed without merging**